### ISSUE TITLE
`print_tree` should accept other kwargs for python's `print` function (in particular, `file` so that printing to a string rather than stdout is possible)

### ISSUE THREAD

**Issue by mmcdermott** [kayjan/bigtree#285]
### Is your feature request related to a problem? Please describe.

I would like to print the tree to a string, not stdout.

### Describe the solution you'd like

I'd like to be able to pass a `file=` arg to `print_tree` which is passed without modification to the `print()` call internally.

### Describe alternatives you've considered

Capturing stdout directly via a context manager.

**Comment by kayjan** [comment]
Hello! Have you considered using yield_tree for more customized usage of print_tree or export in general?

```python
for branch, stem, node in yield_tree(root):
    print(f"{branch}{stem}{node.node_name}")
```

**Comment by mmcdermott** [comment]
Yes, I saw that and considered that, but I don't actually want to work with the tree in a more customized manner -- I just want to print the tree not to stdout but to a file, or a StringIO stream, etc. This functionality exists by default in python, literally all you'd need to do to support this is pass a `file=` kwarg from the print function to the final `print` statement, and this capability would be supported by default. Given how simple a change that is, it seems like it would be worth it to me to support directly, but of course I leave it to your judgement.

**Comment by mmcdermott** [comment]
E.g., see https://github.com/kayjan/bigtree/pull/286/files (though I couldn't get your testing to work so that does need to be validated properly)

**Comment by kayjan** [comment]
Thanks for the PR! It’s still missing test cases let me see if I can add them - for your error with pytest benchmark, can refer to the testing instructions on the documentation 👍

**Comment by kayjan** [comment]
Have deployed this in v0.20.0, thanks for contributing!
